doc/index.rst

```diff
@@ -13,22 +13,44 @@ metrics.
 This package contains efficient Python implementations of several popular
 metric learning algorithms.
 
+Supervised Algorithms
+---------------------
+Supervised metric learning algorithms take as inputs points `X` and target
+labels `y`, and learn to make points from the same class (for classification)
+or with close target value (for regression) close to each other, and points
+from different classes or with distant target values far away from each
+other.
+
 .. toctree::
-   :caption: Algorithms
    :maxdepth: 1
 
    metric_learn.covariance
    metric_learn.lmnn
-   metric_learn.itml
-   metric_learn.sdml
-   metric_learn.lsml
    metric_learn.nca
    metric_learn.lfda
+   metric_learn.mlkr
+
+Semi-Supervised Algorithms
+--------------------------
+Semi-supervised algorithms do the same, but work on a weaker information about
```

Comment:
maybe avoid "do the same" to have a more self consistent section
quick suggestion: "Weakly supervised algorithms work on weaker information about the data points than supervised algorithms. Rather than labeled points, they take as input similarity judgments on tuples of data points, for instance pairs of similar and dissimilar points. Refer to the documentation of each algorithm for its particular form of input data."

Reply:
Yes I agree this is clearer
Done